### ISSUE TITLE
dummy-variables-rgx does not ignore E0602 when its for undefined-variable

### ISSUE THREAD

**Issue by adiel91** [pylint-dev/pylint#9156]
### Bug description

Trying to ignore errors of E0602, Using --dummy-variables-rgx "pattern",
Doesn't ignore errors of type undefined-variable.

```python
import os
def func():
    test_unused = 2
    os.system(test_undefined)
    print("test")

func()
```

* Trying with just test_undefined as the regex pattern, didn't help.

### Command used

```shell
python3 -m pylint --dummy-variables-rgx "test_unused|test_undefined" test.py
python3 -m pylint --dummy-variables-rgx "test_undefined" test.py
```

### Pylint output

```python
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:3:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:5:14: E0602: Undefined variable 'test_undefined' (undefined-variable)

As you can see, pylint still complains about test_undefined
( But not about test_unused - it is properly ignored )
```

### Expected behavior

undfined-variables errors should be ignored as well.

### Pylint version

```shell
pylint 3.0.1
python 3.11.4
```

### OS / Environment

Debian 11 Bullseye

**Comment by adiel91** [comment]
Anyone?

**Comment by Pierre-Sassoulas** [comment]
I don't think dummy-variable-regex is used for that, it's for thing like something that is unused in loop or in a function. If you want to disable a message you should use message control (`` # pylint: disable=undefined-variable``).
